ironfish-cli/src/commands/wallet/multisig/dkg/create.ts

```diff
@@ -0,0 +1,235 @@
+
+export class DkgCreateCommand extends IronfishCommand {
+  static description = 'Interactive command to create a multisignature account using DKG'
+
+  static flags = {
+    ...RemoteFlags,
+    participantName: Flags.string({
+      char: 'n',
+      description: 'The name of the participant to use for DKG',
+    }),
+    accountName: Flags.string({
+      char: 'a',
+      description: 'The name to set for the imported account',
+    }),
+  }
+
+  async start(): Promise<void> {
+    const { flags } = await this.parse(DkgCreateCommand)
+    const client = await this.connectRpc()
+    await ui.checkWalletUnlocked(client)
+
+    const participantName = flags.participantName
+    const accountName = flags.accountName
+
+    const { name, identity } = await this.retryStep('Get or create participant', async () => {
+      return this.getOrCreateParticipant(client, participantName)
+    })
+
+    const { round1Result } = await this.retryStep(
+      'Collect participant info and perform Round 1',
+      async () => {
+        return this.performRound1(client, name, identity)
+      },
+    )
+
+    const { round2Result, round1PublicPackages } = await this.retryStep(
+      'Collect Round 1 public packages and perform Round 2',
+      async () => {
+        return this.performRound2(client, name, round1Result)
+      },
+    )
+
+    await this.retryStep('Perform Round 3', async () => {
+      await this.performRound3(client, name, accountName, round2Result, round1PublicPackages)
+    })
+
+    this.log('DKG process completed successfully!')
+  }
+
+  private async retryStep<T>(stepName: string, stepFunction: () => Promise<T>): Promise<T> {
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      try {
+        this.log(`\nStarting step: ${stepName}`)
+        const result = await stepFunction()
+        this.log(`Step completed: ${stepName}`)
+        return result
+      } catch (error) {
+        this.logger.log(`An error occurred during ${stepName}: ${(error as Error).message}`)
+        await ui.confirmOrQuit('Do you want to retry this step?')
+      }
+    }
+  }
+
+  async getOrCreateParticipant(
+    client: RpcClient,
+    name?: string,
+  ): Promise<{ name: string; identity: string }> {
+    if (!name) {
+      name = await ui.inputPrompt('Enter a name for your participant identity', true)
+    }
+
+    let identity: string
+
+    try {
+      identity = (await client.wallet.multisig.getIdentity({ name })).content.identity
+      this.log(`Using existing participant: ${name}`)
+    } catch (error) {
+      if (
+        error instanceof RpcRequestError &&
+        error.code === RPC_ERROR_CODES.IDENTITY_NOT_FOUND.toString()
+      ) {
+        this.log(`Creating new participant: ${name}`)
+        identity = (await client.wallet.multisig.createParticipant({ name })).content.identity
+      } else {
+        throw error
+      }
+    }
+
+    this.log(`Participant identity: ${identity} for ${name}`)
```

Comment:
nit - i think it's nice to have the identity on its own line to make copy/pasting easier